src/transformers/models/opt/modeling_opt.py

```diff
@@ -700,12 +701,15 @@ def custom_forward(*inputs):
 
                     return custom_forward
 
+                gradient_checkpointing_kwargs = get_checkpointing_kwargs()
+
                 layer_outputs = torch.utils.checkpoint.checkpoint(
                     create_custom_forward(decoder_layer),
                     hidden_states,
                     causal_attention_mask,
                     head_mask[idx] if head_mask is not None else None,
                     None,
+                    **gradient_checkpointing_kwargs,
```

Comment:
I don't know at all if it better or not, but we could wrap the import of and maybe 
`torch.utils.checkpoint.checkpoint = custom_checkpoint`

Reply:
Yes I was thinking of having a util function wrapping `torch.utils.checkpoint.checkpoint` not just the kwargs creation, soory if I was unclear.